src/App.jsx

```diff
@@ -14,33 +15,116 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+export const App = () => {
+  const [goods, setGoods] = useState(goodsFromServer);
+  const [sortedAlph, setSortedAlph] = useState(false);
+  const [sortedByLength, setSortedByLength] = useState(false);
+  const [reverse, setReverse] = useState(false);
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+  const applySortingAndReverse = (array, sortType) => {
+    const updatedGoods = [...array];
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+    if (sortType === 'alphabetical') {
+      updatedGoods.sort((good1, good2) => good1.localeCompare(good2));
+    } else if (sortType === 'length') {
+      updatedGoods.sort((good1, good2) => good1.length - good2.length);
+    }
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+    if (reverse) {
+      updatedGoods.reverse();
+    }
+
+    return updatedGoods;
+  };
+
+  const changeGoods = condition => {
+    switch (condition) {
+      case 'sortAlph':
+        if (sortedAlph) {
+          setSortedAlph(false);
+          setGoods(applySortingAndReverse(goodsFromServer, null));
+        } else {
+          setSortedAlph(true);
+          setSortedByLength(false);
+          setGoods(applySortingAndReverse(goodsFromServer, 'alphabetical'));
+        }
+
+        break;
+
+      case 'sortByLength':
+        if (sortedByLength) {
+          setSortedByLength(false);
+          setGoods(applySortingAndReverse(goodsFromServer, null));
+        } else {
+          setSortedByLength(true);
+          setSortedAlph(false);
+          setGoods(applySortingAndReverse(goodsFromServer, 'length'));
+        }
+
+        break;
+
+      case 'reverse':
+        setReverse(!reverse);
+        setGoods([...goods].reverse());
+        break;
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+      case 'reset':
+        setSortedAlph(false);
+        setSortedByLength(false);
+        setReverse(false);
+        setGoods([...goodsFromServer]);
+        break;
+
+      default:
+        break;
+    }
+  };
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={`button is-info ${sortedAlph ? '' : 'is-light'}`}
```

Comment:
To set classes conditionally it's better to use [classnames](https://www.npmjs.com/package/classnames) or [clsx](https://www.npmjs.com/package/clsx) library